src/test/kotlin/org/jmailen/gradle/kotlinter/functional/ExtensionTest.kt

```diff
@@ -0,0 +1,130 @@
+package org.jmailen.gradle.kotlinter.functional
+
+import groovy.util.GroovyTestCase.assertEquals
+import java.io.File
+import org.gradle.testkit.runner.TaskOutcome
+import org.intellij.lang.annotations.Language
+import org.junit.Before
+import org.junit.Test
+
+internal class ExtensionTest : WithGradleTest() {
+
+    lateinit var projectRoot: File
+
+    @Before
+    fun setUp() {
+        projectRoot = testProjectDir.root.apply {
+            resolve("settings.gradle") { writeText(settingsFile) }
+            resolve("build.gradle") {
+                @Language("groovy")
+                val buildScript = """
+                plugins {
+                    id 'kotlin'
+                    id 'org.jmailen.kotlinter'
+                }
+                
+            """.trimIndent()
+                writeText(buildScript)
+            }
+        }
+    }
+
+    @Test
+    fun `extension configures ignoreFailures`() {
+        projectRoot.resolve("build.gradle") {
+            @Language("groovy")
+            val script = """
+                kotlinter {
+                    ignoreFailures = true
+                }
+            """.trimIndent()
+            appendText(script)
+        }
+        projectRoot.resolve("src/main/kotlin/FileName.kt") {
+            writeText(kotlinClass("DifferentClassName"))
+        }
+
+        build("lintKotlin").apply {
+            assertEquals(TaskOutcome.SUCCESS, task(":lintKotlinMain")?.outcome)
+        }
+    }
+
+    @Test
+    fun `extension configures indentSize`() {
+        projectRoot.resolve("build.gradle") {
+            @Language("groovy")
+            val script = """
+                kotlinter {
+                    indentSize = 2
```

Comment:
I didn't have any good idea how to test if setting `continuationIndentSize`, `experimentalRules` or `fileBatchSize` works as expected.

Reply:
It's ok. When I make changes to those functionalities I specifically test them.

The integration tests are super valuable, but I still also go through manually testing the plugin on ~ 3 sample projects I have that have different complexities. Gradle plugins are devilish to test and inevitably you run into stuff that's hard to cover synthetically like old gradle/kotlin versions, project structure issues.

Reply:
It's ok. When I make changes to those functionalities I specifically test them.

The integration tests are super valuable, but I still also go through manually testing the plugin on ~ 3 sample projects I have that have different complexities. Gradle plugins are devilish to test and inevitably you run into stuff that's hard to cover synthetically like old gradle/kotlin versions, project structure issues.